src/Artifacts.jl

```diff
@@ -314,6 +330,16 @@ function download_artifact(
         msg  = "Tree Hash Mismatch!\n"
         msg *= "  Expected git-tree-sha1:   $(bytes2hex(tree_hash.bytes))\n"
         msg *= "  Calculated git-tree-sha1: $(bytes2hex(calc_hash.bytes))"
+        @static if Sys.iswindows()
+            if !check_symlinks()
+                msg *= """
+
+                Note: Julia cannot create symlinks, which may be the reason for the hash mismatch.
+                You may need to activate Developer Mode in Windows.
+                If you cannot you can set the env var `JULIA_PKG_IGNORE_HASHES=1` to ignore artifact hash mismatches.
```

Comment:
```suggestion
                Note: Julia cannot create symlinks, which may be the reason for the hash mismatch.
                One solution is to activate Developer Mode in Windows, for instructions on how to do that, see:
                    https://learn.microsoft.com/en-us/gaming/game-bar/guide/developer-mode
                An alternative path is to set the environment variable  `JULIA_PKG_IGNORE_HASHES=1`
                to ignore artifact hash mismatches, however this is generally not recommended.
```

Reply:
SGTM but this might be a more appropriate link? https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development

Reply:
I don't like that my link has "gaming" in it, but the link you posted is also really confusing because it doesn't actually tell you how to enable developer mode, the only instructions it gives are for responding to a popup inside of visual studio.  There's no instructions telling you to go to "updates and security" and flip the toggle, that's why I didn't use it.

Also, there are lots of warnings on that site that say things like `If you're not creating your own applications on your PC, you don't need to enable Developer Mode.`, or `If you're using your computer for ordinary day-to-day activities (such as gaming, web browsing, email, or Office apps), you don't need to activate Developer Mode, and in fact, you shouldn't activate it.`.

Reply:
yeah fair

Reply:
If we're going to tell people to set an environment variable on Windows, stating  `ENV["JULIA_PKG_IGNORE_HASHES"] = 1` explicitly, might be good. Otherwise, we will need to get into the details on how to do that in either the Cmd.exe shell or PowerShell.
* Cmd.exe: `set JULIA_PKG_IGNORE_HASHES=1`
* PowerShell: `$env:JULIA_PKG_IGNORE_HASHES=1`

Reply:
We could also add an easier way to do it given we're changing julia in adding this message

Reply:
I tweaked the wording to make it clearer to users and guide them to use ENV so they're more likely to do it temporarily